src/discordcr/websocket.cr

```diff
@@ -51,16 +51,27 @@ module Discord
       end
     end
 
+    ZLIB_SUFFIX = Bytes[0x0, 0x0, 0xFF, 0xFF]
+
+    @zlib_reader : Zlib::Reader?
+    @buffer : Bytes
+
     def initialize(@host : String, @path : String, @port : Int32, @tls : Bool, @logger : Logger)
       @websocket = HTTP::WebSocket.new(
         host: @host,
         path: @path,
         port: @port,
         tls: @tls
       )
+
+      # 10MB buffer for zlib-stream
+      @buffer_memory = Bytes.new(10 * 1024 * 1024)
```

Comment:
It seems unnecessary to allocate this unless we're actually using stream compression, so could you maybe change it to only allocate the buffer in that case? 10MB isn't much in the large scale of things but this seems like an easy way to save some memory.

Reply:
Sure, I can make that happen when the callback is registered 👍